front-end/src/components/AnalyzersAndSeries.js

```diff
@@ -0,0 +1,135 @@
+import '../styles/modalStyles.css';
+
+const AnalyzersAndSeries = ({ dimensions, columns, setDisplayPage, fileTransfer, setShowModal, graphType, liveCheck }) => {
+
+    const columnGenerator = (n) => {
+        let arr = [];
+        for (let i = 0; i < n; i++) {
+            arr.push(
+                <div > <div className="boldText">{i === 0 ? "X-Axis" : "Y-Axis"} </div>
+                    <select className={i} key={i}>
+                        {columns.map(column => (
+                            <option value={JSON.stringify(column)} key={column.header + column.filename}>{column.filename} - {column.header}</option>
+                        ))}
+                    </select>
+                </div>);
+        }
+        return arr;
+    }
+
+    const analyzers = [
+        { name: "None", code: null, parameters: [], checked: true },
+        { name: "Linear Interpolate", code: "linearInterpolate", parameters: [] },
+        { name: "Accel Curve", code: "accelCurve", parameters: [] },
+        { name: "Rolling Average", code: "rollAvg", parameters: [{ name: "WindowSize", default: "100" }] },
+        { name: "RDP Compression", code: "RDPCompression", parameters: [{ name: "Epsilon", default: "0.1" }] },
+        { name: "sGolay", code: "sGolay", parameters: [{ name: "Window Size", default: "100" }, { name: "Polynomial Order", default: "3" }] },
+        { name: "Split", code: "split", parameters: [{ name: "Start", default: "0" }, { name: "End", default: null }] },
+        { name: "Linear Multiply", code: "linearMultiply", parameters: [{ name: "Multiplier", default: "1" }, { name: "Offset", default: "0" }] }
+    ];
+
+    const getAnalyzerOptions = () => {
+        for (const inputElement of analyzers) {
+            if (document.getElementById(inputElement.name).checked) {
+                const params = inputElement.parameters.map(param => {
+                    //console.log(paramId)
+                    const value = document.getElementById(param.name).value;
+                    return value === '' ? null : value;
+                });
+                //console.log(params);
+                return params;
+            }
+        }
+        return null;
+    };
+
+    var seriesInfo = []
+
+    const addSeries = () => {
+
+        var selectColumns = [];
+        for (let i = 0; i < dimensions; i++) {
+            selectColumns.push(JSON.parse(document.getElementsByClassName(i)[0].value));
+        }
+
+        seriesInfo.push({
+            "columns": selectColumns,
+            "analyze": {
+                "analysis": analyzers.filter(analyzer => document.getElementById(analyzer.name).checked)[0].code,
+                "analyzerValues": getAnalyzerOptions()
+            }
+
+        })
+        console.log(seriesInfo)
+    }
+
+    const handleSubmit = (e) => {
```

Comment:
handleFileTransfer should be more fleshed out, so that it only needs to be passed information, and handled submit should no longer be needed. Define in app.js to clarify how information is passed